File: mguide/src/main/java/com/put/mguide/service/AppUserService.java
package com.put.mguide.service;

import org.springframework.stereotype.Service;

import com.put.mguide.model.AppUser;
import com.put.mguide.other.Login;

@Service
public interface AppUserService {
	AppUser login(Login login);
	AppUser register(AppUser user);

}

File: mguide/src/main/ts/mguide/src/app/services/app-user.service.ts
import { Injectable } from '@angular/core';
import { AppUser, Login } from './api';
import { RestService } from './rest.service';

const URL = "api/users";
@Injectable({
  providedIn: 'root'
})
export class AppUserService {

  constructor(private restService: RestService) { }

  login(login: Login) {
    return this.restService.post(`${URL}/login`, login);
  }

  register(user: AppUser) {
    return this.restService.post(`${URL}/register`, user);
  }
}
